### DISCUSSION TITLE
Type 'IDistribution' is missing the following properties from type 'Distribution': domainName, defaultBehavior ....

### DISCUSSION THREAD

**Discussion by dev-samples** [aws/aws-cdk#21733] [Q&A]
I am trying to import an existing distribution with:

```
    const distribution = Distribution.fromDistributionAttributes(scope, 'distribution', {
      distributionId: 'E2***',
      domainName: 'd1u****.cloudfront.net',
    });

    const dummyService = new DummyService(this, 'dummy-service', distribution);

```

where:

```
import { Distribution } from 'aws-cdk-lib/aws-cloudfront';
import { Construct } from 'constructs';

export class DummyService extends Construct {
  constructor(scope: Construct, id: string, distribution: Distribution) {
    super(scope, id);
    const anotherId = distribution.distributionId;
  }
}

```
but it gives this error:

`Type 'IDistribution' is missing the following properties from type 'Distribution': domainName, defaultBehavior, additionalBehaviors, boundOrigins, and 19 more.
`

But the - existing - distribution already has default behavior, domainName, etc. It seems that `fromDistributionAttributes` actually tries to create a new distribution instead or importing the existing one.

What am I missing??

**Comment by cubabit** [comment]
Did you figure this one out? I am having the same problem
